apache/zeppelin
# [ZEPPELIN-1697] Fix multiuser ws connection for secondary ZeppelinHubRepo storage

### What is this PR for?
This is to fix problem with ws connection that was introduced in 0.7.0 with `ZeppelinHubRepo` storage layer. 


### What type of PR is it?
Bug Fix

### Todos
* [x] - fix multi-user ws connection issue
* [x] - handle back connection
* [x] - handle switch tokens
* [x] - handle single token in conf

### What is the Jira issue?
https://issues.apache.org/jira/browse/ZEPPELIN-1697

### How should this be tested?
scenario 1.
1. setup storage as in [here](https://zeppelin.apache.org/docs/0.7.0/storage/storage.html#storage-in-zeppelinhub), note that can use single storage as well (useful when switching instances)
2. setup authentication through ZeppelinHub realm as in [here](https://zeppelin.apache.org/docs/0.7.0/security/shiroauthentication.html#zeppelinhub).
3. login and run paragraph

scenario 2. 
this scenario has basic Zeppelin authentication
1. setup storage as in [here](https://zeppelin.apache.org/docs/0.7.0/storage/storage.html#storage-in-zeppelinhub), note that can use single storage as well (useful when switching instances)
2. use non-authenticated mode
3. start zeppelin and run paragrpaph

other cases would be nice to try and report

### Screenshots (if appropriate)

### Questions:
* Does the licenses files need update? no
* Is there breaking changes for older versions? no
* Does this needs documentation? no
